kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -158,6 +158,22 @@ public static boolean checkLoadBalancerIsWorking(String namespace) {
         return isWorking;
     }
 
+    /**
+     * Wait for deployment to be restarted.
+     *
+     * @param namespaceName the namespace name
+     * @param podName the pod name
+     * @param timeout the timeout
+     */
+    public static void waitForDeploymentToBeRestarted(String namespaceName, String podName, Duration timeout) {
+        await().alias("await pod to be deleted or in non-running phase")
```

Comment:
This is racey.   If the pod is restarted quickly, the client (the test) may never see if pod in the absent or not running state, so the test will fail spuriously.

A safe approch is to rely on the fact that the replacement pod will have a new  `metadata.Uid`.  Record the `uid` before initiating the restart, then await for the pod to exist and have a new `uid`.  I believe that approach will avoid the race condition.

Reply:
This is what I had before Sam's comment:
```
DeploymentUtils.waitForDeploymentRunning(deployNamespace, podName, Duration.ofMinutes(5));
return !Objects.equals(podUid, getPodUid(deployNamespace, podName));
```
We already check for the pod `uid`. So I will leave it as it was.